presto-main/src/test/java/io/prestosql/sql/planner/TestOrExpressionNormalizer.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.sql.planner;
+
+import io.prestosql.metadata.Metadata;
+import io.prestosql.metadata.MetadataManager;
+import io.prestosql.sql.planner.assertions.BasePlanTest;
+import io.prestosql.sql.planner.assertions.SymbolAliases;
+import io.prestosql.sql.tree.SymbolReference;
+import org.testng.annotations.Test;
+
+import static io.prestosql.sql.ExpressionTestUtils.assertExpressionEquals;
+import static io.prestosql.sql.planner.OrExpressionNormalizer.rewrite;
+import static io.prestosql.sql.planner.iterative.rule.test.PlanBuilder.expression;
+
+public class TestOrExpressionNormalizer
+        extends BasePlanTest
+{
+    private static final Metadata METADATA = MetadataManager.createTestMetadataManager();
+
+    @Test
+    public void testSimpleOrExpressions()
+    {
+        assertExpression("x = 0 or x = 1 or x = 2 or x = 3", "x in (0, 1, 2, 3)");
+        assertExpression("(x = 0 or x = 1) or (x = 2 or x = 3)", "x in (0, 1, 2, 3)");
+        assertExpression("x = 0 or (x = 1 or (x = 2 or x = 3))", "x in (0, 1, 2, 3)");
+        assertExpression("x = 0 or (x = 1 or (x = 2 or x = a))", "x in (0, a, 1, 2)");
+        assertExpression("x = 0 or x in (1, 2, 3)", "x in (0, 1, 2, 3)");
+        assertExpression("x > 10 or x in (1, 2, 3)", "x > 10 or x in (1, 2, 3)");
+    }
+
+    @Test
+    public void testOnMultiColumns()
+    {
+        assertExpression("a = 0 or a = 1 or x = 2 or x = 3", "a in (0, 1) or x in (2, 3)");
+        assertExpression("(a = 0 or x = 1) or (a = 2 or x = 3)", "a in (0, 2) or x in (1, 3)");
+        assertExpression("a = 0 or (a = 1 or (x = 2 or x = 3))", "a in (0, 1) or x in (2, 3)");
+        assertExpression("(a = 0 or a = 1) and (x = 2 or x = 3)", "a in (0, 1) and x in (2, 3)");
+    }
```

Comment:
can you please add a test with `x = a or x = b or x IN (2,3,4) or x IN (SELECT 1)`

Also can you please add such tests with actual query using tpch connector to make sure that results are correct.